libraries/rush-lib/src/api/Subspace.ts

```diff
@@ -321,4 +326,66 @@ export class Subspace {
   public _addProject(project: RushConfigurationProject): void {
     this._projects.push(project);
   }
+
+  /**
+   * Returns hash value of injected dependencies in related package.json.
+   * @beta
+   */
+  public getPackageJsonInjectedDependenciesHash(): string | undefined {
+    const allPackageJson: IPackageJson[] = [];
+
+    const relatedProjects: RushConfigurationProject[] = [];
+    const subspacePnpmfileShimSettings: ISubspacePnpmfileShimSettings =
+      SubspacePnpmfileConfiguration.getSubspacePnpmfileShimSettings(this._rushConfiguration, this);
+
+    for (const rushProject of this.getProjects()) {
+      const injectedDependencies: Array<string> =
+        subspacePnpmfileShimSettings?.subspaceProjects[rushProject.packageName]?.injectedDependencies || [];
+      if (injectedDependencies.length === 0) {
+        continue;
+      }
+
+      const injectedDependencySet: Set<string> = new Set(injectedDependencies);
+
+      for (const dependencyProject of rushProject.dependencyProjects) {
+        if (injectedDependencySet.has(dependencyProject.packageName)) {
+          relatedProjects.push(dependencyProject);
+        }
+      }
+    }
+
+    // this means there is no injected dependencies for current subspace
+    if (relatedProjects.length === 0) {
+      return undefined;
+    }
+
+    // recursive to get all related package.json
+    while (relatedProjects.length > 0) {
+      const rushProject: RushConfigurationProject = relatedProjects.pop()!;
+      allPackageJson.push(this._getImportantFieldsInPackageJson(rushProject.packageJsonEditor));
+      relatedProjects.push(...rushProject.dependencyProjects);
+    }
+
+    return objectHasher({ sort: true }).hash(allPackageJson);
+  }
+
+  private _getImportantFieldsInPackageJson(packageJsonEditor: PackageJsonEditor): IPackageJson {
+    const packageJson: IPackageJson = packageJsonEditor.saveToObject();
+
+    // these fields in the package.json even it is modified
+    // it has nothing to do with the pnpm-lock.yaml
```

Comment:
Would it be better to extract/collect the list of properties that are known to affect installation, rather than deleting the properties that are known to be irrelevant?

**package.json** was (unfortunately) designed as a property bag that allows arbitrary keys can be added by various tools without any coordination, namespacing, or rules about where they can appear.  Although it's generally a good practice to be paranoid, in this case I think we can safely assume that Rush can keep up with the relatively small set of keys that impact the `pnpm install` algorithm.

Reply:
Yeah, originally, I was thinking what if there are some new fields in the `package.json` affecting the `pnpm-lock.yaml`, in this case, deleting unrelated fields will be future-proof. However, this also has drawback that new field may not affect `pnpm-lock.yaml`, then it could cause problems. 
So, I am agreeing with you, collecting the list of properties that are known to affect installation will have more control on Rush side, eliminate the uncertainty of `package.json` fields.